src/components/SignInForm/SignInForm.jsx

```diff
@@ -0,0 +1,167 @@
+import { useContext, useState, useCallback } from "react";
+import styles from "./SignInForm.module.scss";
+import { AuthContext } from "../../containers/AuthContext";
+
+const SignInForm = () => {
+  const authContext = useContext(AuthContext);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [errors, setErrors] = useState({});
+
+  const handleEyeButtonClick = (e) => {
+    e.preventDefault();
+    setIsPasswordVisible(!isPasswordVisible);
```

Comment:
this is a potential bug in case if you don't make handleEyeButtonClick as useCallBack(). 
but, we don't actually need useCallBack for it here, there is more simple solution for cases what we need to toggle state of calcite new state value based on its previous value, here: https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state 
always prefer passing callback into `setWhatever` state function.

Reply:
thanks for the useful information! fixed